main.js

```diff
@@ -304,71 +270,59 @@ client.on('message_create', async (message) => {
-            
-                const contentID = await db.getContentID(hash_SHA512);
-                files = convert.filesJSON(ffilename, fmimetype, mFileBlobID, sizeInBytes, sizeInBytes, fSHA512, contentID);
-            }
-            else {
-                console.log("DATA KUCUK!!!");
-                const type = 1;
-                let data = new TextEncoder("utf-8").encode(media_data);
-                data = Buffer.from(data.buffer);
-                await db.createContent_txn(rd_uuidv, data, type, hash_SHA256,
-                          2, 3, mFileBlobID, sizeInBytes, sizeInBytes, hash_SHA512);
-
-                const contentID = await db.getContentID(hash_SHA512);
-                files = convert.filesJSON(ffilename, fmimetype, mFileBlobID, sizeInBytes, sizeInBytes, fSHA512, contentID);
-            }
+        const contentF = await db.contentsAll_txn(rd_uuidv, media_data, 3, 2);
+        if (contentF === null) {
+            files = convert.filesB64JSON(ffilename, fmimetype, media_data);
+        }
+        else {
+            files = convert.filesJSON(ffilename, fmimetype, contentF.reg, contentF.size, contentF.size, fSHA512, contentF.ContentID);
         }
     }
 
-    await db.add_message_txn(message.body, (await message.getChat()).isGroup, ChatID, rd_uuidv, chat.name, message._data.id._serialized, 
-        message.timestamp, convert.senderJSON(message.from, fromName), 
-                    convert.recipientJSON(message.to, toName), files, irtMUUID, mimeQuoted, header, preview, bodyBlob); //database imp demo
+    const MessagesTable = await db.add_message_txn(rd_uuidv, message._data.id._serialized, message.timestamp, sender, recipient, files, irtMUUID, quotedMsg_MIME_ID, header, preview, bodyBlob);
+    await db.folders_txn(chat.id._serialized, chat.name, chat.isGroup, MessagesTable.id, folderUUID);
+    await db.mbody_txn(rd_uuidv, message.body);
+                    
     if (!message.fromMe) {
-        await db.ContactINFO_txn(ContactINF);
+        const ContactID = await db.contacts_txn(contact.id._serialized, contact.name);
+        const response = await axios.get((await contact.getProfilePicUrl()), { responseType: 'arraybuffer' });
+        const AvatarData = Buffer.from(response.data, 'binary');
+        const fSHA512 = crypto.createHash('sha512').update(AvatarData).digest('base64');
+
+        let avatarData;
+    
+        let isContactKnown = false;
+        if (contact.name !== undefined) {
+            isContactKnown = true;
+        }
+        
+        const contentF = await db.contentsAll_txn(rd_uuidv, AvatarData, ContactID, 4);
+        if (contentF === null) {
+            avatarData = "'" + AvatarData.toString('base64') + "'";
+        } else {
+            const JSON_str = [[contentF.reg, String(contentF.size), String(contentF.size), fSHA512]];
+            avatarData = JSON.stringify(JSON_str);
+        }
+    
+        let ContactINF = {
```

Comment:
This contact info isn't consistent with the actual attr keys in ``contactattrs_txn`` function. This won't cause any trouble since this is for logging, but I think this needs to be fixed.